### DISCUSSION TITLE
Custom Components in Create Modal

### DISCUSSION THREAD

**Discussion by entekadesign** [area17/twill#1881] [Q&A] [answered]
Until recently, I was able to use custom vue components in the create.blade modal. Now, it seems that the main-form asset, which contains the compiled custom components, is not getting loaded by the listing layout, which is the layout that is loaded when using the create modal.

I don't want to load the main-form asset as well as the main-listing asset if there is a better way. I see from https://github.com/area17/twill/discussions/1321  that it is possible to create a custom layout that loads the main-form asset outside of a form layout. Is it necessary that I utilize an entire custom layout to re-enable the use of custom components in the create modal, or is there a trick out there that would make my life easier? Thanks for any guidance you can give!

**Comment by ifox** [comment, marked as answer]
Hi @entekadesign, I'm surprised you are saying that it used to work for you because main-listing.js never loaded custom components, we actually have an issue open about it which we're aiming to tackle as soon as 3.0 is out: https://github.com/area17/twill/issues/1314

**Reply by entekadesign** [reply]
Thanks for the super speedy reply!

I'm not sure how it was working previously--I didn't look into it until it ceased doing what I expected. I wonder if it would help to mention that I am using a custom partial in the create blade file, which is linked to the custom vue component--i.e., there are no a17 tags in the create blade file, just a "@formField('custom_partial'..." directive. I was essentially following the pattern described in part 3 here (except applying it to create.blade rather than form.blade): https://twill.io/docs/guides/creating_custom_components_form_fields_and_blocks.html#part-3-custom-form-field-example

*** Edit ***
I figured out how it was working. I hacked the source code--(gasp). I must have forgotten that I modified the main-listing.js, as discussed in https://github.com/area17/twill/discussions/1022 and described in this patch:

https://github.com/pboivin/twill/commit/e97d5e63b4962975d8cd764e6ad23c3e553261b1

I look forward to this feature in 3.0 and thank you for taking the time to answer!
